lnurl.py

```diff
@@ -1,16 +1,77 @@
 from http import HTTPStatus
 
-from fastapi import Request
+from fastapi import Request, Query
 from lnurl import LnurlErrorResponse, LnurlPayActionResponse, LnurlPayResponse
 from starlette.exceptions import HTTPException
 
 from lnbits.core.services import create_invoice
 from lnbits.utils.exchange_rates import get_fiat_rate_satoshis
 
 from . import lnurlp_ext
-from .crud import increment_pay_link
+from .crud import increment_pay_link, get_pay_link, get_address_data
+from loguru import logger
+from urllib.parse import urlparse
 
 
+# for .well-known/lnurlp
+async def lnurl_response(username: str, domain: str, request: Request):
+    address_data = await get_address_data(username)
+
+    if not address_data:
+        return {"status": "ERROR", "reason": "Address not found."}
+
+    resp = {
+        "tag": "payRequest",
+        "callback": request.url_for(
+            "lnurlp.api_lnurl_callback", link_id=address_data.id
+        ),
+        "metadata": await address_data.lnurlpay_metadata(domain=domain),
+        "minSendable": int(address_data.min * 1000),
+        "maxSendable": int(address_data.max * 1000),
+    }
+
+    logger.debug("RESP", resp)
+    return resp
+
+
+# for lnaddress callback
+@lnurlp_ext.get(
+    "/api/v1/lnurl/cb/{link_id}",
+    status_code=HTTPStatus.OK,
+    name="lnurlp.api_lnurl_callback",
+)
+async def api_lnurl_callback(request: Request, link_id, amount: int = Query(...)):
+    address = await get_pay_link(link_id)
+    if not address:
+        return LnurlErrorResponse(reason=f'{"Address not found"}').dict()
+
+    domain = urlparse(str(request.url)).netloc
+    assert domain
+
+    unhashed_description = await address.lnurlpay_metadata(domain=domain)
+    unhashed_description = unhashed_description.encode()
```

Comment:
Type changes mid-flight. Better: 
```suggestion
    unhashed_description = await address.lnurlpay_metadata(domain=domain).encode()
```